src/Symfony/Component/HttpKernel/DependencyInjection/RemoveEmptyControllerArgumentLocatorsPass.php

```diff
@@ -47,8 +47,7 @@ public function process(ContainerBuilder $container)
             } else {
                 // any methods listed for call-at-instantiation cannot be actions
                 $reason = false;
-                $action = substr(strrchr($controller, ':'), 1);
-                $id = substr($controller, 0, -1 - strlen($action));
+                list($id, $action) = explode('::', $controller);
```

Comment:
What about using `[$id, $action]` here?

Reply:
We could but I don't think we use that syntax in the code base yet.